src/widgets/TemperatureSyncState.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { StyleSheet, Text, TouchableOpacity } from 'react-native';
+
+import { TemperatureSyncIcon } from './TemperatureSyncIcon';
+
+import globalStyles from '../globalStyles';
+
+import { TemperatureSyncActions } from '../actions/TemperatureSyncActions';
+import {
+  selectTemperatureSyncStateMessage,
+  selectIsSyncingTemperatures,
+} from '../selectors/temperatureSync';
+
+const mapStateToProps = state => {
+  const syncMessage = selectTemperatureSyncStateMessage(state);
+  const isSyncing = selectIsSyncingTemperatures(state);
+
+  return { syncMessage, isSyncing };
+};
+
+const mapDispatchToProps = dispatch => {
+  const onOpenModal = () => dispatch(TemperatureSyncActions.openModal());
+  return { onOpenModal };
+};
+
+export const TemperatureSyncStateComponent = ({ isSyncing, onOpenModal, syncMessage }) => (
```

Comment:
Pure?

Reply:
As above don't need to when you `connect`